internal/server/audit/template/executer.go

```diff
@@ -0,0 +1,86 @@
+package template
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"strings"
+	"text/template"
+	"time"
+
+	"go.flipt.io/flipt/internal/server/audit"
+	"go.uber.org/zap"
+)
+
+// webhookTemplate contains fields that pertain to constructing an HTTP request.
+type webhookTemplate struct {
+	method string
+	url    string
+
+	headers      map[string]string
+	bodyTemplate *template.Template
+
+	maxBackoffDuration time.Duration
+
+	retryableClient audit.Retrier
+}
+
+// Executer will try and send the event to the configured URL with all of the parameters.
+// It will send the request with the retyrable client.
+type Executer interface {
+	Execute(context.Context, audit.Event) error
+}
+
+// NewWebhookTemplate is the constructor for a WebhookTemplate.
+func NewWebhookTemplate(logger *zap.Logger, method, url, body string, headers map[string]string, maxBackoffDuration time.Duration) (Executer, error) {
+	tmpl, err := template.New("").Parse(body)
+	if err != nil {
+		return nil, err
+	}
+
+	return &webhookTemplate{
+		url:                url,
+		method:             method,
+		bodyTemplate:       tmpl,
+		headers:            headers,
+		maxBackoffDuration: maxBackoffDuration,
+		retryableClient:    audit.NewRetrier(logger, maxBackoffDuration),
+	}, nil
+}
+
+func (w *webhookTemplate) createRequest(ctx context.Context, body []byte) (*http.Request, error) {
```

Comment:
could you add a test for this method, covering the missing test cover messages below?